File: examples/import.c
/*
 *  import        libcaca importers test program
 *  Copyright (c) 2006-2012 Jean-Yves Lamoureux <jylam@lnxscene.org>
 *                All Rights Reserved
 *
 *  This program is free software. It comes without any warranty, to
 *  the extent permitted by applicable law. You can redistribute it
 *  and/or modify it under the terms of the Do What the Fuck You Want
 *  to Public License, Version 2, as published by Sam Hocevar. See
 *  http://www.wtfpl.net/ for more details.
 */

#include "config.h"

#if !defined(__KERNEL__)
#   include <stdio.h>
#   include <stdlib.h>
#endif

#include "caca.h"

int main(int argc, char *argv[])
{
    caca_canvas_t *cv;
    caca_display_t *dp;

    if(argc < 2)
    {
        fprintf(stderr, "%s: missing argument (filename).\n", argv[0]);
        fprintf(stderr, "usage: %s <filename> [<format>]\n", argv[0]);
        return 1;
    }

    cv = caca_create_canvas(0, 0);
    if(cv == NULL)
    {
        printf("Can't create canvas\n");
        return -1;
    }

    if(caca_import_canvas_from_file(cv, argv[1], argc >= 3 ? argv[2] : "") < 0)
    {
        fprintf(stderr, "%s: could not open `%s'.\n", argv[0], argv[1]);
        caca_free_canvas(cv);
        return 1;
    }

    dp = caca_create_display(cv);
    if(dp == NULL)
    {
        printf("Can't create display\n");
        return -1;
    }

    caca_refresh_display(dp);

    caca_get_event(dp, CACA_EVENT_KEY_PRESS, NULL, -1);

    caca_free_display(dp);
    caca_free_canvas(cv);

    return 0;
}

File: caca-php/examples/import.php
#!/usr/bin/php5
<?php
/*
 *  import        libcaca importers test program
 *  Copyright (c) 2008 Benjamin C. Wiley Sittler <bsittler@gmail.com>
 *
 *  This file is a Php port of "examples/import.c"
 *  which is:
 *  Copyright (c) 2006 Jean-Yves Lamoureux <jylam@lnxscene.org>
 *                All Rights Reserved
 *
 *  This program is free software. It comes without any warranty, to
 *  the extent permitted by applicable law. You can redistribute it
 *  and/or modify it under the terms of the Do What the Fuck You Want
 *  to Public License, Version 2, as published by Sam Hocevar. See
 *  http://www.wtfpl.net/ for more details.
 */

if (php_sapi_name() != "cli") {
    die("You have to run this program with php-cli!\n");
}

$imports = caca_get_import_list();

if($argc < 2 || $argc > 3)
{
    $msg = ($argv[0] . ": wrong argument count\n" .
            "usage: " . $argv[0] . " file [<format>]\n" .
            "where <format> is one of:\n");
    foreach($imports as $format => $name)
        $msg .= " \"" . $name . "\" (" . $format . ")\n";
    die($msg);
}

$cv = caca_create_canvas(0, 0);
if(! $cv)
{
    die("Can't create canvas\n");
}

if(caca_import_file($cv, $argv[1], $argc >= 3 ? $argv[2] : "") < 0)
{
    die($argv[0] . ": could not open `" . $argv[1] . "'.\n");
}

$dp = caca_create_display($cv);
if(! $dp)
{
    die("Can't create display\n");
}

caca_refresh_display($dp);

caca_get_event($dp, CACA_EVENT_KEY_PRESS, -1);

?>
